interfaces/ValuePoint.h

```diff
@@ -312,7 +316,27 @@ namespace Exchange {
     protected:
         inline void Updated() const
         {
-            _job.Submit();
+            if (_clients.size() > 0) {
+                notify metadata = notify::METADATA;
+                notify none     = notify::NONE;
+
+                if ((_notify.compare_exchange_strong(none, notify::UPDATE) == true) ||
+                    (_notify.compare_exchange_strong(metadata, static_cast<notify>(notify::METADATA | notify::UPDATE)) == true)) {
+                    _job.Submit();
+                }
+            }
+        }
+        inline void Metadata() const
+        {
+            if (_clients.size() > 0) {
+                notify update = notify::UPDATE;
+                notify none   = notify::NONE;
+
+                if ((_notify.compare_exchange_strong(none, notify::METADATA) == true) ||
+                    (_notify.compare_exchange_strong(update, static_cast<notify>(notify::METADATA | notify::UPDATE)) == true)) {
```

Comment:
Think there is a race condition possible here, suppose the compare(none,...) above failed because update() was called before. Now just before this line (336) notify was called from the job, the line  mode = _notify.exchange called there only returning mode to be update and mode set to none, now this line will fail because mode is no longer update and the mode in notify will not indicate the meta needs to be handles and it will be missed (basically the two compare's are not atomic as a pair, the exchange in notify can happen inbetween them

Reply:
Wanted to avoid the CriticalSection Lock but it feels there is no way around it :-) Will adapt the code. Good feedback!